medicines/api/Dockerfile

```diff
@@ -0,0 +1,41 @@
+FROM rust:1.40 AS build
+
+WORKDIR /usr/src/api
+
+COPY Cargo.toml .
+COPY Cargo.lock .
+
+# Layer hack: Build an empty program to compile dependencies and place on their own layer.
+# This cuts down build time
+RUN mkdir -p ./src/ && \
+  echo 'fn main() {}' > ./src/main.rs && \
+  echo '' > ./src/lib.rs
```

Comment:
Is this best practice? Was this recommended somewhere? Interested to know how this works.